src/pl/ogladajanime/src/eu/kanade/tachiyomi/animeextension/pl/ogladajanime/OgladajAnime.kt

```
@@ -0,0 +1,293 @@
+
+    override fun videoListParse(response: Response): List<Video> {
+        val jsonResponse = json.decodeFromString<ApiResponse>(response.body.string())
+        val dataObject = json.decodeFromString<ApiData>(jsonResponse.data)
+        val serverList = mutableListOf<Pair<String, String>>()
+
+        dataObject.players.forEach { player ->
+            var sub = player.sub.uppercase()
+            if (sub == "PL" && player.audio == "PL") {
+                sub = "DUB PL"
+            }
+
+            val prefix = if (player.ismy > 0) {
+                "[$sub/Odwrócone Kolory] "
+            } else {
+                "[$sub] "
+            }
+
+            if (player.url in listOf("vk", "cda", "mp4upload", "sibnet", "dailymotion")) {
+                val url = getPlayerUrl(player.id)
+                serverList.add(Pair(url, prefix))
+            }
+        }
+        // Jeśli dodadzą opcje z mozliwością edytowania mpv to zrobić tak zejak bedą odwrócone kolory to ustawia dane do mkv <3
+        val videoList = serverList.flatMap { (serverUrl, prefix) ->
+            when {
+                serverUrl.contains("vk.com") -> {
+                    vkExtractor.videosFromUrl(serverUrl, prefix)
+                }
+                serverUrl.contains("mp4upload") -> {
+                    mp4uploadExtractor.videosFromUrl(serverUrl, headers, prefix)
+                }
+                serverUrl.contains("cda.pl") -> {
+                    cdaExtractor.getVideosFromUrl(serverUrl, headers, prefix)
+                }
+                serverUrl.contains("dailymotion") -> {
+                    dailymotionExtractor.videosFromUrl(serverUrl, "$prefix Dailymotion -")
+                }
+                serverUrl.contains("sibnet.ru") -> {
+                    sibnetExtractor.videosFromUrl(serverUrl, prefix)
+                }
+                else -> emptyList()
+            }
+        }
+        return videoList.sort()
```

```suggestion
        return serverList.parallelCatchingFlatMapBlocking { (serverUrl, prefix) ->
            when {
                serverUrl.contains("vk.com") -> {
                    vkExtractor.videosFromUrl(serverUrl, prefix)
                }
                serverUrl.contains("mp4upload") -> {
                    mp4uploadExtractor.videosFromUrl(serverUrl, headers, prefix)
                }
                serverUrl.contains("cda.pl") -> {
                    cdaExtractor.getVideosFromUrl(serverUrl, headers, prefix)
                }
                serverUrl.contains("dailymotion") -> {
                    dailymotionExtractor.videosFromUrl(serverUrl, "$prefix Dailymotion -")
                }
                serverUrl.contains("sibnet.ru") -> {
                    sibnetExtractor.videosFromUrl(serverUrl, prefix)
                }
                else -> emptyList()
            }
        }
```

Using `parallelCatchingFlatMapBlocking ` will try to extract videos in parallel, thus improving load times. It will also ignore any errors caused by extractors, so the video will still play if something goes wrong somewhere